src/org/thoughtcrime/securesms/ConversationActivity.java

```diff
@@ -1538,7 +1538,9 @@ public void handleEvent(DcEvent event) {
       dcChat = dcContext.getChat(chatId);
       titleView.setTitle(glideRequests, dcChat);
       initializeSecurity(isSecureText, isDefaultSms);
-      invalidateOptionsMenu();
+      if (searchMenu == null) {
+        invalidateOptionsMenu();
```

Comment:
Couldn't this check be moved into `onPrepareOptionsMenu`, so every time `invalidateOptionsMenu` is called, it actually updates the menu to the correct state? Then it will not be necessary to avoid calling `invalidateOptionsMenu` on the caller side.

This check looks rather like a workaround for incorrect behavior of `onPrepareOptionsMenu`.

Reply:
in private conversations we found out that `onPrepareOptionsMenu` is not the reason the menu is collapsed

Reply:
maybe @Hocuri can tell

Reply:
`onPrepareOptionsMenu()` is called everytime the user opens the options menu, anyway. What `invalidateOptionsMenu()` does, is trigger a new call to `onCreateOptionsMenu()`. Source: https://developer.android.com/reference/android/app/Activity.html#invalidateOptionsMenu()

Just wondering why we need to call `invalidateOptionsMenu()` at all, if we do everything in `onPrepareOptionsMenu()` anyway, so that the menu is updated everytime the user opens it.

Reply:
Apart, I wonder why we call `invalidateOptionsMenu()` in `handleEvent()` specifically, as the menu can't change at `DC_EVENT_CHAT_MODIFIED`, `DC_EVENT_CHAT_EPHEMERAL_TIMER_MODIFIED` or `DC_EVENT_CONTACTS_CHANGED`, except if I missed something.

The only reason that the options menu can change inside a chat I can think of is, the chat is muted ("Mute notifications"->"Unmute") and "Unarchive"->"Archive". And `invalidateOptionsMenu()` isn't even called at either of these, which is no problem, because we update the complete menu everytime it's opened, anyway.

All in all, I think you could just remove all `invalidateOptionsMenu()` calls from `ConversationActivity`.

Reply:
One problem is if you are removed from the group, "Leave group" menu should be changed.

Reply:
> I think it just needs to be tested after the change, that when you are removed from the group and open menu immediately after that, there is no "Leave group" option.

Actually, currently, the "Leave group" option never disappears, but that's another issue. Also, `can_send()` should probably return false if you are not in the group, so that the input panel disappears.

Reply:
my first impression was also that all this was unnecessary